# Agentchat API with Assistant Agent

In [1]:
from autogen_agentchat.agents import AssistantAgent

In [2]:
# model
from autogen_ext.models.openai import OpenAIChatCompletionClient
from dotenv import load_dotenv
load_dotenv()
import os
api_key =  os.getenv("OPEN_ROUTER_API_KEY")

In [12]:
model_client = OpenAIChatCompletionClient(
    base_url="https://openrouter.ai/api/v1",
    api_key=api_key,
    model="deepseek/deepseek-r1-0528:free",
    model_info={
        "family":'deepseek',
        "vision" :True,
        "function_calling":True,
        "json_output": False
    }
)

In [13]:
my_assistant = AssistantAgent(name="Assistant", model_client=model_client)

In [15]:
result = await my_assistant.run(task="who are you?")
print(result)

messages=[TextMessage(id='031c444f-f74e-48fd-96f9-0aa0942c0ac3', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2026, 2, 10, 17, 12, 56, 344973, tzinfo=datetime.timezone.utc), content='who are you?', type='TextMessage'), TextMessage(id='a87d4cbe-1f15-414f-ba8a-0cac199021df', source='Assistant', models_usage=RequestUsage(prompt_tokens=88, completion_tokens=426), metadata={}, created_at=datetime.datetime(2026, 2, 10, 17, 13, 17, 718353, tzinfo=datetime.timezone.utc), content="I am an AI assistant created to provide helpful, accurate responses and assist with tasks using various tools. You can ask me questions, request information, or seek help—I'll do my best to respond clearly and efficiently!\n\nIs there something specific you'd like to know or work on? 😊", type='TextMessage')] stop_reason=None


messages=[TextMessage(id='031c444f-f74e-48fd-96f9-0aa0942c0ac3', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2026, 2, 10, 17, 12, 56, 344973, tzinfo=datetime.timezone.utc), content='who are you?', type='TextMessage'), 

TextMessage(id='a87d4cbe-1f15-414f-ba8a-0cac199021df', source='Assistant', models_usage=RequestUsage(prompt_tokens=88, completion_tokens=426), metadata={}, created_at=datetime.datetime(2026, 2, 10, 17, 13, 17, 718353, tzinfo=datetime.timezone.utc),

 content="I am an AI assistant created to provide helpful, accurate responses and assist with tasks using various tools. You can ask me questions, request information, or seek help—I'll do my best to respond clearly and efficiently!\n\nIs there something specific you'd like to know or work on? 😊", type='TextMessage')] stop_reason=None

In [19]:
result = await my_assistant.run(task="what was my last question I asked?")
print(result)

messages=[TextMessage(id='aa81de35-9389-49fa-8525-bcff4ebd3870', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2026, 2, 10, 17, 16, 49, 337099, tzinfo=datetime.timezone.utc), content='what was my last question I asked?', type='TextMessage'), TextMessage(id='e77dfdb0-efa6-41c6-9ccd-1809e8810155', source='Assistant', models_usage=RequestUsage(prompt_tokens=216, completion_tokens=723), metadata={}, created_at=datetime.datetime(2026, 2, 10, 17, 17, 26, 420330, tzinfo=datetime.timezone.utc), content='Your last question was: **"what was my last question I asked?"**  \n\nTo avoid recursion, here\'s the full sequence:  \n1. Your first question: "who are you?"  \n2. Repeated: "who are you?"  \n3. Your **previous question**: "what was my last question I asked?" (this is the one I\'m responding to now).  \n\nWould you like to discuss something else? 😊', type='TextMessage')] stop_reason=None


messages=[TextMessage(id='aa81de35-9389-49fa-8525-bcff4ebd3870', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2026, 2, 10, 17, 16, 49, 337099, tzinfo=datetime.timezone.utc), content='what was my last question I asked?', type='TextMessage'), 

TextMessage(id='e77dfdb0-efa6-41c6-9ccd-1809e8810155', source='Assistant', models_usage=RequestUsage(prompt_tokens=216, completion_tokens=723), metadata={}, created_at=datetime.datetime(2026, 2, 10, 17, 17, 26, 420330, tzinfo=datetime.timezone.utc), content='Your last question was: **"what was my last question I asked?"**  \n\nTo avoid recursion, here\'s the full sequence:  \n1.

 Your first question: "who are you?"  \n2. Repeated: "who are you?"  \n3. Your **previous question**: "what was my last question I asked?" (this is the one I\'m responding to now).  \n\nWould you like to discuss something else? 😊', type='TextMessage')] stop_reason=None

In [ ]:
result = await my_assistant.run(task="what is the capital of India and what is 4 + 4?")
print(result)